spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisInboundChannelAdapter.java

```diff
@@ -75,6 +77,11 @@ public void setMessageConverter(MessageConverter messageConverter) {
 		this.messageConverter = messageConverter;
 	}
 
+	public void setTaskExecutor(Executor taskExecutor) {
+		Assert.notNull(messageConverter, "taskExecutor must not be null");
```

Comment:
`taskExecutor` can be `null`. See the logic in the `RedisMessageListenerContainer.afterPropertiesSet()`.
Although you see: the story here should not be about `messageConverter`.
After removing this `Assert` I'm sure the Checkstyle violation will go by itself:
```
:spring-integration-redis:checkstyleMain[ant:checkstyle] [ERROR] /home/travis/build/spring-projects/spring-integration/spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisInboundChannelAdapter.java:81:32: Reference to instance variable 'messageConverter' needs "this.". [RequireThis]
```